web/api/maestro_api/db/repo/run_metric.py

```diff
@@ -64,18 +64,22 @@ def group_metrics(self, run_id, time_interval, show_labels):
             "run_id": ObjectId(run_id),
         }
 
-        group_id = {
-            "year": {"$year": "$datetime"},
-            "dayOfYear": {"$dayOfYear": "$datetime"},
-            "hour": {"$hour": "$datetime"},
-            "minute": {"$minute": "$datetime"},
-            "interval": {
-                "$subtract": [
-                    {"$second": "$datetime"},
-                    {"$mod": [{"$second": "$datetime"}, time_interval]},
-                ]
-            },
-        }
+        group_id = (
+            {
+                "year": {"$year": "$datetime"},
+                "dayOfYear": {"$dayOfYear": "$datetime"},
+                "hour": {"$hour": "$datetime"},
+                "minute": {"$minute": "$datetime"},
+                "interval": {
+                    "$subtract": [
+                        {"$second": "$datetime"},
+                        {"$mod": [{"$second": "$datetime"}, time_interval]},
+                    ]
+                },
+            }
+            if time_interval > 0
```

Comment:
@dyegowermelinger Can you also add a test for this condition?

Reply:
Test created in the commit: d1fcb24295ec048b53092366faa540d4ac93c892